### PR TITLE
fix half-measures

### PR DESCRIPTION
**Pull request by cscherrer**
TODO
- [x] Add tests

### CODE REVIEW

**Comment by cscherrer** [comment]
@sethaxen I think it's fixed, but the fix requires MeasureBase v0.3.7, which is pending registration. I assume we'll need to re-run the tests once that's ready.

File: src/parameterized/studentt.jl [hunk 1]
```diff
@@ -1,17 +1,20 @@
...
+@kwstruct StudentT(ν,μ)
+@kwstruct StudentT(ν,σ)
+@kwstruct StudentT(ν,μ,σ)
 
-StudentT(nt::NamedTuple{(:ν,:μ,:σ)}) = Affine(nt, StudentT())
-StudentT(nt::NamedTuple{(:ν,:μ,:ω)}) = Affine(nt, StudentT())
-StudentT(nt::NamedTuple{(:ν,:σ,)}) = Affine(nt, StudentT())
-StudentT(nt::NamedTuple{(:ν,:ω,)}) = Affine(nt, StudentT())
-StudentT(nt::NamedTuple{(:ν,:μ,)}) = Affine(nt, StudentT())
+StudentT(nt::NamedTuple{(:ν,:μ,:σ)}) = Affine(NamedTuple{(:μ,:σ)}(nt), StudentT(ν=nt.ν))
```
Comment (sethaxen) [inline src/parameterized/studentt.jl:13]: I know this isn't part of this PR, but overloading constructors to construct something else can cause problems, the simplest being that something like `StudentT(0, 1, 30) isa StudentT` is `false`.

File: test/runtests.jl [hunk 2]
```diff
@@ -396,3 +396,33 @@ end
...
+    end
+
+    @testset "HalfCauchy" begin
+        d = Cauchy(σ=3)
+        h = HalfCauchy(3)
+        x = rand(h)
+        @test density(h, Lebesgue(ℝ), x) ≈ 2 * density(d, Lebesgue(ℝ), x)
+    end
+
+    @testset "HalfLaplace" begin
```
Comment (sethaxen) [inline test/runtests.jl:415]: Isn't the half-Laplace just the Exponential distribution?

**Review by sethaxen — APPROVED** [review]
Pending tests passing, this PR LGTM. I did have some extra-PR questions.

File: test/runtests.jl (hunk 2 above)
Comment (cscherrer) [reply test/runtests.jl:415]: Oh weird, hadn't ever thought of that. That sounds right, if it is we can just drop `HalfLaplace`

File: src/parameterized/studentt.jl (hunk 1 above)
Comment (cscherrer) [reply src/parameterized/studentt.jl:13]: Yeah, I agree that's a little weird. But the upside is pretty huge. This approach allows us to fine-tune code for `Affine`, and have that same code work for multivariate-anything. So implementing MvNormal gives us MvStudentT etc for free.

To me it seems worth it, but if you disagree we should talk through it

Comment (cscherrer) [reply src/parameterized/studentt.jl:13]: Made this an issue:
https://github.com/cscherrer/MeasureTheory.jl/issues/157

I'll resolve this now, we can discuss there

**Pull request merged**